### PR TITLE
Generate WebVTT Files

### PR DESCRIPTION
**Pull request by raian621**
Added a `format` parameter to the `/transcribe/` API endpoint that enables the generation of either `vtt` caption files or `json` transcript data. Also added a new endpoint, `/generate-vtt/` that accepts VTT specific data like CSS styles and caption positioning options as per the specifications for options, styles, and cues listed here: https://developer.mozilla.org/en-US/docs/Web/API/WebVTT_API.

The frontend has no way of calling the `/generate-vtt/` endpoint or the `/transcribe/` endpoint with the `format=vtt` parameter right now, so you have to test the VTT generation by using the server's FastAPI Swagger documentation page at `localhost:8000/docs`.

### CODE REVIEW

**Comment by raian621** [comment]
oh shoot I need to fix something on the client side I manually changed

**Comment by raian621** [comment]
don't merge it yet

File: server/models/web_vtt.py [hunk 1]
```diff
@@ -0,0 +1,250 @@
...
+
+# any integer or percentage between 0% and 100% inclusive:
+line_regex = re.compile(r"^-?\d+$|^100%$|^[1-9]?[0-9]%$")
+# any percentage between 0% and 100% inclusive:
+clamped_percent_regex = re.compile(r"^100%$|^[1-9]?[0-9]%")
+# css class / id name
+css_class_id_regex = re.compile(r"^[A-Za-z_].*$")
+
+# reference: https://www.tutorialrepublic.com/css-reference/css3-properties.php
+CSS_PROPERTIES = [
```
Comment (nomaddevs1) [inline server/models/web_vtt.py:15]: The styling is being handled by the frontend all you had to do is just generate the vtt and send it to the frontend. We would worry about the cue styling

File: server/tests/fixtures/vtt/advanced_data_with_styles.vtt
Comment (nomaddevs1) [inline server/tests/fixtures/vtt/advanced_data_with_styles.vtt:1]: Same styling is being handled by the frontend

**Review by nomaddevs1 — CHANGES_REQUESTED** [review]
This suppose to be on the forked repo not this

**Comment by nickingenito** [comment]
> This suppose to be on the forked repo not this

Since its already in and just changing the backend its probably fine to just add it here and then sync the forked repo

**Comment by nomaddevs1** [comment]
> > This suppose to be on the forked repo not this
> 
> Since its already in and just changing the backend its probably fine to just add it here and then sync the forked repo

Yh that works. The changes for the styling is what is left

**Comment by raian621** [comment]
Closing PR on this repo, moving it to the other repo

File: server/models/web_vtt.py (hunk 1 above)
Comment (raian621) [reply server/models/web_vtt.py:15]: The styling here is completely optional, and is only there for future extensibility :) .

You can style the captions on the frontend, fine, but the client is may want the ability to download custom WebVTT captions in the future.

**Pull request closed without merging**